src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -492,9 +509,36 @@ private Tags tags() {
         return null;
     }
 
-    private Scheduling scheduling() {
+    @VisibleForTesting
+    Scheduling scheduling() {
         Scheduling scheduling = new Scheduling();
-        scheduling.setPreemptible(preemptible);
+        long maxRunDurationSeconds = 0;
+        if (provisioningType != null) { // check `null` for backward compatibility
+            ProvisioningTypeValue ptValue = provisioningType.getValue();
+            if (ptValue == PREEMPTIBLE) {
```

Comment:
All these `if then else` statements doesn't make the provisioning type very extensible.

For example you could instead have the provisioning type configure the scheduling itself:

`provisioningType.configure(scheduling)`

and then in for example `SpotVm`

```java
 @Override
public void configure(Scheduling s) {
  s.setProvisioningModel("SPOT");
  super.configure(s) //would for example set maxRunDurationSeconds if needed
}
```

Reply:
already fixed this one, thanks for suggestion, makes sense.
https://github.com/jenkinsci/google-compute-engine-plugin/pull/492/commits/3e7e7872cb2989de02b2201ab92c8e9d27931281